Add Size & SizeF structs

wolfgang-ironsoftware opened the pull request:
### Title
Add Size & SizeF structs and modify Rectangle & RectangleF to have constructors that make use of these structs.

### Description
Adds the Size & SizeF structs (somewhat simplified) from SixLabors.

### Fixes #(issue number)
Doesn't fix any particular issue, but will be useful in IronPrint to specify sizes for printing.

### Type of change
Please select the relevant option by placing an 'x' inside the brackets, like this: [x].

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🏗️ Internal/structural update (non-breaking change that improves code quality, organization, or performance)
- [x] 📚 This change requires a documentation update
- [ ] 🚀 DevOps build chain modification for release
- [ ] 🤖 DevOps build chain modification for CI

### How Has This Been Tested?
Added unit tests from SixLabors.

### Checklist:
Please run through the checklist as much as possible and mark the items completed by placing an 'x' inside the brackets, like this: [x].

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have successfully run all unit tests on Windows
- [x] I have successfully run all unit tests on Linux

### Additional Context
Add any other context, screenshots, or information about the pull request here.

mee-ironsoftware reviewed (commented):
@wolfgang-ironsoftware 
Could you add implicit cast for another library I saw only `SixLabors.ImageSharp`?
We need `System.Drawing`, `SkiaSharp`, and `Microsoft.Maui.Graphics`.

wolfgang-ironsoftware commented:
> @wolfgang-ironsoftware
> Could you add implicit cast for another library I saw only `SixLabors.ImageSharp`?
> We need `System.Drawing`, `SkiaSharp`, and `Microsoft.Maui.Graphics`.

Good idea! I'll add Monday.

wolfgang-ironsoftware commented:
@mee-ironsoftware Added implicit converters for other libraries you mentioned.

Also added to SizeF struct (which didn't include them).

Also added VS studio generated `GetHashCode()` functions as required by the `IEquatable` interface.

wolfgang-ironsoftware commented:
@mee-ironsoftware Added more operators just a second ago. So have conversion both to and from various types.

mee-ironsoftware reviewed (approved):
Nice one Thanks @wolfgang-ironsoftware
